pkg/operator/ceph/object/objectstore.go

```diff
@@ -845,13 +846,20 @@ func poolName(poolPrefix, poolName string) string {
 }
 
 // GetObjectBucketProvisioner returns the bucket provisioner name appended with operator namespace if OBC is watching on it
-func GetObjectBucketProvisioner(data map[string]string, namespace string) string {
+func GetObjectBucketProvisioner(data map[string]string, namespace string) (string, error) {
 	provName := bucketProvisionerName
 	obcWatchOnNamespace := k8sutil.GetValue(data, "ROOK_OBC_WATCH_OPERATOR_NAMESPACE", "false")
-	if strings.EqualFold(obcWatchOnNamespace, "true") {
+	obcProvisionerNamePrefix := k8sutil.GetValue(data, "ROOK_OBC_PROVISIONER_NAME_PREFIX", "")
```

Comment:
If you pass the `namespace` as the default value, then you can simplify the block below.
```suggestion
	obcProvisionerNamePrefix := k8sutil.GetValue(data, "ROOK_OBC_PROVISIONER_NAME_PREFIX", namespace)
```

Reply:
following case won't handled if I pass default value , then ROOK_OBC_PROVISIONER_NAME_PREFIX not set and "ROOK_OBC_WATCH_OPERATOR_NAMESPACE" set false. Then ideally it should return `ceph.rook.io/bucket` but here it will return `rook-ceph.ceph.rook.io/bucket`